src/dune_lang/lib_kind.ml

```diff
@@ -63,13 +63,15 @@ type t =
   | Ppx_deriver of Ppx_args.t
   | Ppx_rewriter of Ppx_args.t
   | Parameter
+  | Virtual
```

Comment:
IIUC, an even more robust encoding would represent this as `Virtual of <virtual_modules>` however that is a more invasive change, and I think would call for deprecating the existing syntax for `virtual_modules`. This could be something we pursue as a followup, but I think should not be included here, as it doesn't help meaningfully with advancing https://github.com/ocaml/dune/issues/12085, which is the focus here.